### PR TITLE
Refactoring - Theme & FavouriteCities states

### PR DESCRIPTION
**Pull request by Lvlad27**
- Theme and FavouriteCities states are now implemented through Redux and persisted in Local Storage
- Made some minor styling changes
- Remove unnecessary .env options

### CODE REVIEW

File: src/Components/Header/ThemeSwitch/ThemeSwitch.jsx [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
+              id="checkbox-toggle"
+              onClick={() => dispatch(toggleTheme())}
+            />
+          ) : (
+            <input
+              type="checkbox"
+              defaultChecked={false}
+              id="checkbox-toggle"
+              onClick={() => dispatch(toggleTheme())}
+            />
```
Comment (JuliaShavyrova) [inline src/Components/Header/ThemeSwitch/ThemeSwitch.jsx]: why not like this? or just make this input controlled
```
<input
  type="checkbox"
  checked={theme === "light"}
  id="checkbox-toggle"
  onClick={() => dispatch(toggleTheme())}
/>
```

File: src/redux/index.js [hunk 2]
```diff
@@ -1,48 +1,24 @@
...
 
-const persistedState = loadState();
-const store = createStore(cityReducer, persistedState);
+const store = createStore(rootReducer, persistedState);
 
 store.subscribe(() => {
-  saveState(store.getState());
+  saveState('savedLocations', store.getState().favCityList);
+  saveState('theme', store.getState().theme);
+  console.log(store.getState());
```
Comment (JuliaShavyrova) [inline src/redux/index.js]: please clean code

File: src/context/themeContext.js [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
+    theme: 'light'
   };
 
   toggleTheme = () => {
-    const darkOS = window.matchMedia('(prefers-color-scheme: dark)').matches;
-
     this.setState((prevState) => {
       return {
-        theme: prevState.theme === 'light' ? 'dark' : 'light'
+        theme: prevState.theme === 'dark' ? 'light' : 'dark'
```
Comment (JuliaShavyrova) [inline src/context/themeContext.js]: do you still need theme provider after you created theme reducer?

Comment (Lvlad27) [reply src/context/themeContext.js]: fixed

File: src/Components/Header/ThemeSwitch/ThemeSwitch.jsx (hunk 1 above)
Comment (Lvlad27) [reply src/Components/Header/ThemeSwitch/ThemeSwitch.jsx]: ok, fixed

File: src/redux/index.js (hunk 2 above)
Comment (Lvlad27) [reply src/redux/index.js]: done

File: src/hooks/useApi.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  useEffect(() => {
+    tokenCheck();
+    findMyLocation(setCoords);
+  }, []);
+
+  useEffect(() => {
+    getCurrentWeatherByCoords();
+    getLocationDataByCoords();
+    getDailyWeather();
+    getHourlyWeather();
```
Comment (JuliaShavyrova) [inline src/hooks/useApi.js]: if you want to get all responses at once, you can try axios.all:
```
axios.all([
  instance.get(endpoints.LOCATION_DATA_BY_COORDS(coords)), 
  instance.get(endpoints.HOURLY_WEATHER(coords)), 
  ....
])
```

Comment (Lvlad27) [reply src/hooks/useApi.js]: done

File: src/hooks/useApi.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+          setCurrWeather(res[0].data);
+          setDailyWeather(res[1].data);
+          setHourlyWeather(res[2].data);
+          setLocationData(res[3].data);
+          setIsLoading(false);
+          setError(null);
+        })
+      )
+      .catch((error) => {
+        setError(error);
```
Comment (JuliaShavyrova) [inline src/hooks/useApi.js:42]: you need to set loading=false here if you don't want to get infinite loading status :)

Comment (Lvlad27) [reply src/hooks/useApi.js:42]: Yes, actually I put it right after the commit. Fixed now

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**